### DISCUSSION TITLE
V2 Roadmap

### DISCUSSION THREAD

**Discussion by hardy613** [GameBridgeAI/ts_serialize#147] [General]
# Recap
1. v0.0.1 -> v1.0.0 was about building the main module
2. v1.0.1 -> v2.0.0 was about solidifying our core code base, documentation and resolving core issues from those using ts_serialize in production
3. v2.0.1 -> v3.0.0 strategies and decorators? 

# v2 
We have a good starting foundation with our current strategies, some get used commonly enough that we can start to provide decorators.

Decorators will need to use [`getNewSerializable`](https://github.com/GameBridgeAI/ts_serialize/blob/develop/strategy/utils.ts#L9]) when dealing with types. We will also want to provide a convenient approach to overriding `tsTransformKey` with a string or function. We'll also want a naming convention that fits our current approach.

To start we can add some decorator idea's, even if strategies will be needed. Below is a shortilst of ideas that we can later expand into a to do list of strategies and decorators

- [ ] Timestamp() - (with UTC options?)
- [ ] ObjectContaining(serializable) - from an object of { k: serializable } -> { k: JSONObject | string } and back
- [ ] Serializable(type) - standard to from serializable types
- [ ] ArrayContaining(serializable) - possible multiple types

other ideas and chores
- [ ] update documentation (readme)
- [ ] `examples/` dir with concret examples (we'll need a testing strat)

**Comment by hardy925** [comment]
Swagger support on keys that have been transformed would be amazing.
